react-ui/src/components/DetailsPage.js

```diff
@@ -1,19 +1,28 @@
 import React, { Component } from 'react';
+import '../styles/DetailsPage.css';
 
 export class DetailsPage extends Component {
   constructor(props) {
     super();
   }
 
-  componentDidMount() {
+  componentWillMount() {
     const { id } = this.props.match.params;
-
     this.props.fetchDetail(id);
   }
 
   render() {
-    const { name } = this.props.locationDetails.location;
-    const happyHours = this.props.locationDetails.happyhours.map(hours => {
+    if (!this.props.locationDetails.location) {
+      return <p>Loader</p>;
+    }
+
+    const hours = this.props.locationDetails.location.data[0];
+    const location = this.props.locationDetails.location.data[1];
```

Comment:
I think if you changed the above comment and got rid of the object literal for data you could remove the dot notation for data here. Also, this seems like a good place to have the reducer handle assignment to hours and location. That way our redux store will be more verbose in naming our objects.